### PR TITLE
Add new Proton Nuclear Information Actor

### PR DESCRIPTION
**Pull request by SimonRit**
See OpenGATE/GateContrib#23 for an example

### CODE REVIEW

File: source/digits_hits/src/GateProtonNuclearInformationActor.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#include "GateMiscFunctions.hh"
+
+// G4
+#include <G4Event.hh>
+#include <G4ParticleTable.hh>
+#include <G4TransportationManager.hh>
+#include "G4HadronicProcessStore.hh"
+#include "G4HadronicProcess.hh"
+
+#define TRY_AND_EXIT_ON_ITK_EXCEPTION(execFunc)                         \
```
Comment (djboersma) [inline source/digits_hits/src/GateProtonNuclearInformationActor.cc]: You define a macro here, but I do not see where you are actually using it. Am I missing something, or is this maybe a leftover from a previous version, and these lines could be removed?

Comment (SimonRit) [reply source/digits_hits/src/GateProtonNuclearInformationActor.cc]: Leftover it seems. Sorry for not taking the time to fully clean up Cath's code. I have removed this macro. Thanks!

File: source/digits_hits/src/GateProtonNuclearInformationActor.cc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+{
+  GateVActor::Construct();
+  //  Callbacks
+  EnableUserSteppingAction(true);
+}
+//-----------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+// Callbacks
+void GateProtonNuclearInformationActor::UserSteppingAction(const GateVVolume * v,
```
[the remark this answers is not in the document]
Comment (djboersma) [reply source/digits_hits/src/GateProtonNuclearInformationActor.cc]: But if this actor is actually not specific for protons, then maybe "GateNuclearInformationActor" would be a more correct name, maybe?

[the remark this answers is not in the document]
Comment (SimonRit) [reply source/digits_hits/src/GateProtonNuclearInformationActor.cc]: Currently, it does check for nuclear processes of protons only ("hadElastic" and "protonInelastic"). I don't know if that would work for other particles but my bet is that it requires some adaptations.

**Pull request merged**